### ISSUE TITLE
Release into ROS Melodic

### ISSUE THREAD

**Issue by 130s** [ros-drivers/freenect_stack#48]
I noticed that freenect_stack depends on [`libfreenect` ROS package](http://wiki.ros.org/libfreenect), which doesn't seem to be released into ROS Melodic yet. It would be great if that's taken care of.

I'm aware of https://github.com/ros-drivers/freenect_stack/issues/23 (a release request for ROS Jade, which is EoL already by now), and https://github.com/ros-drivers/freenect_stack/pull/30 (looks like code update to make compatible to ROS Jade?), but on Travis CI it seems to be building on ROS Kinetic on 16.04 https://github.com/ros-drivers/freenect_stack/pull/47 (e.g. in [this series of build](https://travis-ci.org/ros-drivers/freenect_stack/builds/510257770), only the job for ROS Melodic fails due to missing libfreenect ROS package).

Lastly if it makes sense I can help making ROS distro release (I might be only limited to releasing tasks. I might not be able to review PRs.). Please add me write access to this repo, release repo, and libfreenect if needed.

Thank you.

**Comment by 130s** [comment]
Btw this blocks e.g. https://github.com/turtlebot/turtlebot/issues/272

**Comment by lukewebdev** [comment]
This is an issue for me too; the latest turtlebot branch for melodic has been changed to use openni instead of freenect. I can't seem to find driversfor openni that work on the rockpro64 board I've installed melodic turtlebot branch on, and as a result I can't user my sensor.

**Comment by lukewebdev** [comment]
Incidentally, the package seems to work fine as-is if I manually clone the repository and use it that way. So hopefully it isn't a big undertaking to get this released.

**Comment by luisdomjr11** [comment]
I've manually cloned the libfreenect package, but I still cannot get freenect to build. Did you clone in it in a specific folder? Any help is appreciated. My and my group members are running into this issue on out project, it's a pain.

**Comment by WyattAutomation** [comment]
I was able to install this, and I don't know if it's going to work like I'm expecting but I now have libfreenect and the ros freenect_launch and other freenect_node stuff all installed in Melodic on Ubuntu18.04, per the commands below:

Compile/install libfreenect from source, *NOT* an apt install of libfreenect-dev (if you installed that via apt it shouldn't hurt, it just won't work.  No need to remove that apt package afaik):
```
cd  ~
git clone https://github.com/OpenKinect/libfreenect.git
cd libfreenect
mkdir build
cd build
cmake -L ..
make
sudo make install
```

Then, cd into your catkin workspace source, (I'm assuming yours is "~/catkin_workspace/src" but you may need to change that if yours is different).  Clone the repo, then cd up one directory back into your catkin workspace root, and run catkin_make:
```
cd ~/catkin_workspace/src
git clone https://github.com/ros-drivers/freenect_stack.git
cd ..
catkin_make
```
I'm on Ubuntu 18.04 running ROS Melodic, and was able to get this installed the day of this post.

Whoever is running this repo really should just update the package, it already works, it's just grief for us noobs.. 

(just kidding, I appreciate the work that's been put into this; without it my $8 kinect from a thrift store wouldn't be of any use to me.  Would really like to help make this easier to install on Melodic!!).

**Comment by WyattAutomation** [comment]
Thanks Luke, for at least giving me some hope it's still working in Melodic; literally the only comment on the internet regarding this that wasn't just like "it's impossible"..

Let me know if you run into problems with this; it's still a very useful tool.  It's not 2019 if you can't build robots of doom out of scrap consumer electronics..

**Comment by elgarbe** [comment]
Hi, so you @WyattAutomation get kinect for xbox 360 working ok with this driver?
what is your computer setup? I will try it on odroid xu4 and on my laptot PC.

**Comment by maxsvetlik** [comment]
The main issue is that `libfreenect` is no longer maintained by `ros-drivers` and the install location of the header files for the deb `libfreenect` is different than the install location of the header files for `ros-drivers`'s `libfreenect`.

This is addressed by #51. If you use that fork and install `libfreenect-dev` through `apt`, `freenect-stack` should not complain. With this resolved, hopefully a melodic release can come quickly.

**Comment by maxsvetlik** [comment]
Closed by #51. The `master` branch now targets `melodic` development. `kinetic` development is now on the `kinetic` branch.

**Comment by 130s** [comment]
@maxsvetlik Thanks for taking a lead on this!

> Closed by #51. The `master` branch now targets `melodic` development. `kinetic` development is now on the `kinetic` branch.

I see `master` is still the default branch on this. If you mean the `master` stores only the released state (like `git flow`), I think changing the default to a development branch would avoid confusion.

That said I have no idea how many developers still clone the code of this project, so this may not matter.

**Comment by WyattAutomation** [comment]
@maxsvetlik Thanks for you efforts!
